aggregator/src/main/java/com/redhat/cloud/notifications/DailyEmailAggregationJob.java

```diff
@@ -72,18 +87,19 @@ public void processDailyEmail() {
 
         try {
             LocalDateTime now = LocalDateTime.now(UTC);
-            List<AggregationCommand> aggregationCommands = null;
-            if (featureFlipper.isAggregatorOrgPrefEnabled()) {
-                aggregationOrgConfigRepository.createMissingDefaultConfiguration(defaultDailyDigestTime);
-                aggregationCommands = processAggregateEmailsWithOrgPref(now, registry);
-            } else {
-                aggregationCommands = processAggregateEmails(now, registry);
-            }
+
+            aggregationOrgConfigRepository.createMissingDefaultConfiguration(defaultDailyDigestTime);
+            List<AggregationCommand> aggregationCommands = processAggregateEmailsWithOrgPref(now, registry);
+
             List<CompletableFuture<Void>> futures = new ArrayList<>();
             for (AggregationCommand aggregationCommand : aggregationCommands) {
                 try {
                     final String payload = objectMapper.writeValueAsString(aggregationCommand);
-                    futures.add(emitter.send(payload).toCompletableFuture());
+                    if (featureFlipper.isAggregatorSendOnIngress()) {
+                        sendIt(aggregationCommand);
+                    } else {
+                        futures.add(emitter.send(payload).toCompletableFuture());
+                    }
```

Comment:
I wonder why we initially implemented the aggregation job using `CompletableFutures` rather than relying on `Emitter#send(Message)` as you did here. Maybe there was a reason, maybe not... Anyway it would be better to confirm that calling `Emitter#send(Message)` will actually lead to messages being sent without any interference from the Quarkus shutdown at the end of the CronJob execution.

Reply:
IIRC It was something about quarkus finishing before the messages were sent.

Reply:
> IIRC It was something about quarkus finishing before the messages were sent.

That makes sense when `Emitter#send(String)` is called because that's an async operation (as implied by the `CompletableFuture` return type). On the other hand, `Emitter#send(Message)` is a sync operation and I'm not sure we ever tried using it.